org/mozilla/jss/pkcs11/PK11Signature.c

```diff
@@ -48,39 +55,79 @@ Java_org_mozilla_jss_pkcs11_PK11Signature_initSigContext
     SGNContext *ctxt=NULL;
     jobject contextProxy=NULL;
     SECKEYPrivateKey *privk;
+    SECAlgorithmID *signAlg=NULL;
+    SECStatus rv;
+    int keyTypeOKForPSS = 0;
 
     /* Extract the private key from the PK11Signature */
     if( getPrivateKey(env, this, &privk) != PR_SUCCESS) {
         PR_ASSERT( (*env)->ExceptionOccurred(env) != NULL);
         goto finish;
     }
+    
+    // See if the key is a candidate for PSS signature 
+    if (privk->keyType == rsaKey || privk->keyType == rsaPssKey) {
```

Comment:
Why not pull this into a macro instead of a variable? Seems weird, since you only use it once.

Reply:
Got rid of the variable. Upon further review this is a redundant check, we only care if the alg is correct.